main.tf

```
@@ -8,15 +8,15 @@ locals {
 
 locals {
   _name       = var.use_fullname ? module.this.id : module.this.name
-  image_names = length(var.image_names) > 0 ? var.image_names : [local._name]
+  image_names = length(var.image_names) > 0 ? var.image_names : tomap(local._name)
   repository_creation_enabled = module.this.enabled && var.repository_creation_enabled
 }
 
 resource "aws_ecr_repository" "name" {
-  for_each             = toset(local.repository_creation_enabled ? local.image_names : [])
-  name                 = each.value
+  for_each             = local.repository_creation_enabled ? local.image_names : {}
+  name                 = each.key
   image_tag_mutability = var.image_tag_mutability
-  force_delete         = var.force_delete
+  force_delete         = try(each.value.force_delete_override, false)
```

```suggestion
  force_delete         = each.value.force_delete_override
```

Don't think the try statement is necessary since you already declared the force_delete_override attribute optional

---

I had the same thought. For some reason when running locally I got an error that the map objects of the repos were null, which led me to add `try()`. Is there a better way to test it or just remove `try()`, merge, and then publish 1.0.1 if it doesn't work? 

```terraform
╷
│ Error: Attempt to get attribute from null value
│ 
│   on .terraform/modules/ecr/main.tf line 20, in resource "aws_ecr_repository" "name":
│   20:   force_delete         = each.value.force_delete_override
│     ├────────────────
│     │ each.value is null
│ 
│ This value is null, so it does not have any attributes.
```

---

I think that it's because if you have something like...
```yaml
ecr:
  vars:
    images:
      some_image_0:
        force_delete: true
      some_image_1:
      some_image_2: {}
```

then 

`each.value (some_image_0)` will be `{ force_delete: true }`
`each.value (some_image_1)` will be `null`
`each.value (some_image_2)` will be `{}`

And note that `{}` is _not the same as_ `null`
```shell
$ terraform console
> {} == null
false
```

---

Thank you! That clearly explains some of the discrepancies I saw with my tests. 

Added `{}` to the catalog file in the infrastructure repo and the test plan was successful.